packages/ts/src/components/bullet-legend/modules/shape.ts

```diff
@@ -50,52 +51,68 @@ export function updateBullets (
   container.each((d, i, els) => {
     const shape = getString(d, config.bulletShape, i) as BulletShape
     const color = getColor(d, colorAccessor, i)
-    const width = BULLET_SIZE
+    const colorValue = d.color ?? color
+    const colors = Array.isArray(colorValue) ? colorValue : [colorValue ?? '#000']
```

Comment:
Use can use the `ensureArray` function here.

Also I'm not sure about the fallback to `#000`, I think the color will be automatically defined by CSS if the color value is missing, what do you think?